VSRAD.Package/Commands/QuickActionCommand.cs

```diff
@@ -0,0 +1,99 @@
+        {
+            _project = project;
+            _actionLogger = actionLogger;
+            _channel = channel;
+            _serviceProvider = serviceProvider;
+            _statusBar = new VsStatusBarWriter(serviceProvider);
+        }
+
+        public Guid CommandSet => Constants.QuickActionsMenuCommandSet;
+
+        public OLECMDF GetCommandStatus(uint commandId, IntPtr commandText)
+        {
+            if (commandId == Constants.ProfileCommandId || commandId == Constants.DisassembleCommandId || commandId == Constants.PreprocessCommandId)
+                return OLECMDF.OLECMDF_ENABLED | OLECMDF.OLECMDF_SUPPORTED;
+            return 0;
+        }
+
+        public void Execute(uint commandId, uint commandExecOpt, IntPtr variantIn, IntPtr variantOut)
+        {
+            string commandName;
+            switch (commandId)
+            {
+                case Constants.PreprocessCommandId:
+                    commandName = "Preprocess";
+                    break;
+                case Constants.ProfileCommandId:
+                    commandName = "Profile";
+                    break;
+                case Constants.DisassembleCommandId:
+                    commandName = "Disassemble";
+                    break;
+                default:
+                    commandName = string.Empty;
+                    break;
+            }
+
+            var action = Actions.FirstOrDefault(a => a.Name == commandName);
+
+            if (action == null)
+            {
+                Errors.ShowWarning($"Action {commandName} is not defined. To create it, go to Tools -> RAD Debug -> Options and edit your current profile.");
+                return;
+            }
+
+            VSPackage.TaskFactory.RunAsyncWithErrorHandling(() => ExecuteActionAsync(action));
+        }
+
+        private async Task ExecuteActionAsync(ActionProfileOptions action)
+        {
+            try
+            {
+                await _statusBar.SetTextAsync("Running " + action.Name + " action...");
+
+                var evaluator = await _project.GetMacroEvaluatorAsync().ConfigureAwait(false);
+                var env = await _project.Options.Profile.General.EvaluateActionEnvironmentAsync(evaluator);
+                action = await action.EvaluateAsync(evaluator, _project.Options.Profile);
+
+                var runner = new ActionRunner(_channel, _serviceProvider, env);
+                var result = await runner.RunAsync(action.Name, action.Steps, Enumerable.Empty<BuiltinActionFile>()).ConfigureAwait(false);
+                var actionError = await _actionLogger.LogActionWithWarningsAsync(action.Name, result).ConfigureAwait(false);
+                if (actionError is Error e1)
+                    Errors.Show(e1);
+            }
+            finally
+            {
+                await _statusBar.ClearAsync();
+            }
+        }
+    }
+}
```

Comment:
Here is the repetition of the same function as `ActionsMenuCommand`. It might be better to add some abstraction for `ICommandHandler` that works with `VsStatusBarWriter`

Reply:
Addressed in db4cd27 and 04edcbd